app/controllers/carts_controller.rb

```diff
@@ -10,7 +10,19 @@ def create
     redirect_to request.referer || root_path
   end
 
-  def edit; end
+  def update
+    old_quantity = @cart[params[:session][:product_id]]
+    new_quantity = params.dig(:session, :quantity).to_i
```

Comment:
nếu `params.dig(:session, :quantity)` mà nil thì `.to_i` vẫn thở chứ chú  
để an toàn, 1 là chú return, 2 là chú auto set thành 1 nhé 

`new_quantity = (params.dig(:session, :quantity) || 1).to_i`

Reply:
Dạ chị.